www/PLUG/pagefunctions.inc.php

```diff
@@ -1,14 +1,19 @@
 <?php
 
-if(!isset($pagestarttime)) // For pages that don't need auth
+class UnauthorisedException extends RuntimeException
```

Comment:
While this is old code, we should try and stick with modern best practices. Classes should be defined in their own files